pkg/util/iptables/save_restore_test.go

```diff
@@ -20,19 +20,19 @@ import (
 	"testing"
 
 	"github.com/lithammer/dedent"
+
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-func checkChains(t *testing.T, save []byte, expected map[Chain]struct{}) {
+func checkChains(t *testing.T, save []byte, expected sets.Set[Chain]) {
 	chains := GetChainsFromTable(save)
```

Comment:
Hi @danwinship, @dims suggested I look over some of your network PRs as a way to learn and get familiar with the codebase. Should [GetChainsFromTables](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/iptables/save_restore.go#L43) be checking `end` here instead of `i`?

Reply:
ha! yes. feel free to submit a PR fixing that...

Reply:
Thanks! Let me know if you have any other suggestion of small fixes to go in with this: https://github.com/kubernetes/kubernetes/pull/122093